### ISSUE TITLE
Attached .eml is often rejected by the recipient

### ISSUE THREAD

**Issue by LanceHaverkamp** [justreportit/thunderbird#29]
Attached .eml is often rejected by the recipient. Then we have to open the email, extract the text, paste the text into the email, resend the email.  

I suggest that we stop using a .eml attachment, and instead send the entire text in the body of the abuse email 

Thanks,
Lance

**Comment by nicprov** [comment]
@LanceHaverkamp what type of error are you getting from the recipient? And who is the recipient in your case?

The reason that the email is attached as an .eml is to allow multiple emails to be reported at the same time (as requested in #11).

**Comment by LanceHaverkamp** [comment]
The automatic messages were all variations of: Attachments are not accepted.
I received variation of that same message from nearly all of the first several companies I reported to.

**Comment by nicprov** [comment]
@LanceHaverkamp the main intended recipient of the attachment is SpamCop which supports attachments. 

I'm not sure which service/company you are reporting to, but if they are refusing attachments, that is an issue on their side.

**Comment by RobbyZ** [comment]
+1 on this, having trouble even with spamcop:

I had a devil of a time today getting spamcop to accept messages.  It turns out that using the built-in message forwarding results in stripping headers somewhere between my ISPs and SpamCop.

I did a bunch of testing this afternoon, and here's what I found in case it helps anyone else:

| Method                                            | Result                                                                                                      |
|---------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| Simple forward                                    | NG - No headers included (expected)                                                                         |
| Drag & drop forward                               | NG - Headers stripped by ISP (even though sent folder contains the message as being sent with full headers) |
| Just Report It Button                             | NG - Same as above (headers stripped by ISP)                                                                |
| Copy/paste message source and drop in new message | OK with spamcop (had a long delay though maybe just a one off issue)                                        |
| Simple forward + copy/paste message headers       | OK with spamcop (medium delay)                                                                              |

So basically, I'd like an option to do simple forward, with a brief message like 'I am reporting the following unsolicited spam', then the original message headers in text format, followed by the original message inline.

**Comment by nicprov** [comment]
@RobbyZ Which version of Thunderbird are you using? I was also experiencing a similar issue at one point, but it went away when I upgraded to the latest version of Thunderbird (91.11.0). Can you make sure you are using the latest version?

**Comment by RobbyZ** [comment]
I've just verified this is still happening in `91.11.0 (64-bit)` (which according to the software check is currently latest build in release channel).

The stripping of old 'Received' headers may or may not be a Thunderbird feature -- it could also just be something my mail server does.  

IMO a simple forward method is the most likely method to be widely supported by all software in the reporting chain (thunderbird -> my mail server -> recipient mail server -> abuse contact support ticket software).  If any one of those links mangles attachments, the report fails.  Simple header forward as plaintext seems much more durable IMO.

**Comment by nicprov** [comment]
@RobbyZ is this issue still present in 102.6.0? In any case though, I'm not sure what can be done. I am using the _getRaw(messageId)_ api call as shown [here](https://webextension-api.thunderbird.net/en/stable/messages.html#getraw-messageid).

Here's the description of the function:

> Returns the unmodified source of a message as a [binary string](https://developer.mozilla.org/en-US/docs/Web/API/DOMString/Binary), which is a simple series of 8-bit values.

This therefore seems to be an issue related to Thunderbird. Because of that, I will close this issue, but if you find a solution to this, please open a new issue.
